launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -154,9 +165,17 @@
 
   List<String> buildSparkSubmitArgs() {
     List<String> args = new ArrayList<>();
-    SparkSubmitOptionParser parser = new SparkSubmitOptionParser();
+    OptionParser parser = new OptionParser(false);
+    boolean isStartingApp = isAppResourceReq;
+
+    // If the user args array is not empty, we need to parse it to detect exactly what
+    // the user is trying to run, so that checks below are correct.
+    if (!userArgs.isEmpty()) {
+      parser.parse(userArgs);
+      isStartingApp = parser.isAppResourceReq;
```

Comment:
I don't really care whether the name is `isStartingApp` or `isAppResourceReq`, but seems the name should be same here and in OptionParser, unless there is some difference I'm missing.